apps/launcher/graphicspage.cpp

```diff
@@ -185,6 +213,52 @@ void Launcher::GraphicsPage::saveSettings()
     {
         mEngineSettings.setFloat("framerate limit", "Video", 0);
     }
+
+    int cShadowDist = shadowDistanceCheckBox->checkState() ? shadowDistanceSpinBox->value() : 0;
+    if (mEngineSettings.getInt("maximum shadow map distance", "Shadows") != cShadowDist)
+        mEngineSettings.setInt("maximum shadow map distance", "Shadows", cShadowDist);
+    float cFadeStart = fadeStartSpinBox->value();
+    if (cShadowDist > 0 && mEngineSettings.getFloat("shadow fade start", "Shadows") != cFadeStart)
+        mEngineSettings.setFloat("shadow fade start", "Shadows", cFadeStart);
+
+    bool cActorShadows = actorShadowsCheckBox->checkState();
+    bool cObjectShadows = objectShadowsCheckBox->checkState();
+    bool cTerrainShadows = terrainShadowsCheckBox->checkState();
+    bool cPlayerShadows = playerShadowsCheckBox->checkState();
+    if (cActorShadows || cObjectShadows || cTerrainShadows || cPlayerShadows)
```

Comment:
Not sure why it's done like this, but when all types of shadows are enabled to disable all shadows one need to uncheck all types. That's kind of annoying. Having `enable shadows` setting it would be better just to control it directly without any special logic like this.